src/task.c

```diff
@@ -568,7 +568,7 @@ JL_DLLEXPORT jl_task_t *jl_new_task(jl_function_t *start, jl_value_t *completion
     t->prio = -1;
     t->tid = -1;
 #ifdef ENABLE_TIMINGS
-    t->timing_stack = NULL;
+    t->timing_stack = jl_root_timing;
```

Comment:
I'm not 100% sure about this change, but otherwise we don't pop the root timing block when exiting directly from a task (e.g. calling `exit` from the REPL).